### PR TITLE
Low-level logic to use hardlinks

### PR DESCRIPTION
**Pull request by duboism**
This PR adds the low-level logic to use hardlinks in the configuration files and in low-level functions. By default, hardlinks are disabled so everything is backward-compatible.

### CODE REVIEW

File: biomaj_core/utils.py [hunk 1]
```diff
@@ -305,8 +336,31 @@ def copy_files_with_regexp(from_dir, to_dir, regexps, move=False, lock=None):
...
+                    try:
+                        os.link(from_file, to_file)
+                        logger.debug("Using hardlinks to copy %s",
+                                     file_to_copy['name'])
+                    except OSError as e:
+                        if e.errno == errno.EPERM:
+                            msg = "The FS at %s doesn't support hard links. Using regular copy."
+                            logger.warn(msg, to_dir)
+                            # Copy this file
+                            shutil.copyfile(from_file, to_file)
```
Comment (osallou) [inline biomaj_core/utils.py:351]: Shutil copystat is missing

File: biomaj_core/utils.py [hunk 2]
```diff
@@ -305,8 +336,31 @@ def copy_files_with_regexp(from_dir, to_dir, regexps, move=False, lock=None):
...
+                            # Don't try links anymore
+                            use_hardlinks = False
+                        if e.errno == errno.EXDEV:
+                            msg = "Cross device hard link are impossible (source: %s, dest: %s). Using regular copy."
+                            logger.warn(msg, from_file, to_dir)
+                            # Copy this file
+                            shutil.copyfile(from_file, to_file)
```
Comment (osallou) [inline biomaj_core/utils.py:360]: Shutil copystat is missing too

**Comment by osallou** [comment]
Please also update changes.txt and incrzase version in setup.py

**Comment by abretaud** [comment]
Cool :+1:

File: biomaj_core/utils.py (hunk 1 above)
Comment (duboism) [reply biomaj_core/utils.py:351]: Done in a66b64f.

File: biomaj_core/utils.py (hunk 2 above)
Comment (duboism) [reply biomaj_core/utils.py:360]: Done in a66b64f.

**Comment by duboism** [comment]
> Please also update changes.txt and incrzase version in setup.py

Done in 770a076.

**Review by osallou — APPROVED** [review]

**Comment by duboism** [comment]
@osallou Thanks. We will submit changes in other parts of BioMAJ quickly.

**Pull request merged**